### PR TITLE
[Merged by Bors] - feat(SuccPred): {succ,pred}_{min,max}

### PR DESCRIPTION
**Pull request by pechersky**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/SuccPred/Basic.lean [hunk 1]
```diff
@@ -1443,6 +1443,26 @@ variable [LinearOrder α]
...
+  cases le_total x y with
+  | inl h =>
+    rw [min_eq_left h, min_eq_left]
+    simp [h]
+  | inr h =>
+    rw [min_eq_right h, min_eq_right]
+    simp [h]
+
+theorem succ_max [SuccOrder α] (x y : α) :
+    succ (max x y) = max (succ x) (succ y) := by
```
Comment (eric-wieser) [inline Mathlib/Order/SuccPred/Basic.lean:1457]: @YaelDillies, is this true as
```
theorem succ_min [SemilatticeSup α] [SuccOrder α] (x y : α) :
    succ (x ⊔ y) = succ x ⊔ succ y := by
```

Comment (YaelDillies) [reply Mathlib/Order/SuccPred/Basic.lean:1457]: No

Consider, three copies of `Nat`, two are incomparable and the third one is greater than both. Then `a in bottom left copy ⊔ b in bottom right copy = 0 in top copy`, so
* `succ (0 in bottom left copy) ⊔ succ (0 in bottom right copy) = 1 in bottom left copy ⊔ 1 in bottom right copy = 0 in top copy`
* `succ (0 in bottom left copy ⊔ 0 in bottom right copy) = succ (0 in top copy) = 1 in top copy`.

File: Mathlib/Order/SuccPred/Basic.lean [hunk 2]
```diff
@@ -1443,6 +1443,26 @@ variable [LinearOrder α]
...
 
+theorem succ_min [SuccOrder α] (x y : α) :
+    succ (min x y) = min (succ x) (succ y) := by
+  cases le_total x y with
+  | inl h =>
+    rw [min_eq_left h, min_eq_left]
+    simp [h]
+  | inr h =>
+    rw [min_eq_right h, min_eq_right]
+    simp [h]
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/Basic.lean:1454]:
```suggestion
  succ_mono.map_max _ _
```

**Review by YaelDillies — APPROVED** [review]
Golfed. Thanks!

maintainer merge

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**